actionpack/lib/action_controller/railtie.rb

```diff
@@ -133,5 +133,18 @@ class Railtie < Rails::Railtie # :nodoc:
         ActionController::TestCase.executor_around_each_request = app.config.active_support.executor_around_test_case
       end
     end
+
+    initializer "action_controller.escape_json_responses_deprecated_warning" do
+      config.after_initialize do
+        ActiveSupport.on_load(:action_controller) do
+          if ActionController::Base.escape_json_responses
+            ActionController.deprecator.warn(<<~MSG.squish)
+              Setting action_controller.escape_json_responses = true is deprecated and will have no effect in Rails 8.2.
+              Set it to `false` or use `config.load_defaults(8.1)`.
```

Comment:
It seems odd to recommend `load_defaults` here. Changing that will influence a lot of other things, not just this setting. We don't normally recommend this in deprecation warnings, is there a reason that it's done here?

Reply:
@byroot ^

Reply:
nvm commented heree https://github.com/rails/rails/pull/54840